src/api/store/deviceprofile.py

```diff
@@ -51,6 +44,7 @@ def create_device(self, context: Context, args, save=True) -> Tuple[dict, MassEn
     try:
       new_device: DeviceProfile = DeviceProfile.objects.create(**args)
       
+      # question: is this needed?  those values not already part of new_device through create with the args?
```

Comment:
The __device_attr_handler() method is just a way of reducing duplicate code. It's technically not needed, but having lots of duplicate code is considered a bad practice.

If the checks in this method aren't needed we could remove it. When I wrote it I was mimicking another api store.